src/backend/aspen/api/utils/phylo.py

```diff
@@ -134,43 +172,55 @@ async def _set_countries(
         .select_from(origin_location)  # type: ignore
         .join(
             Location,  # type: ignore
-            and_(
-                Location.division == None,  # noqa: E711
-                Location.location == None,  # noqa: E711
-                Location.country.in_(defined_countries),
-            ),
+            and_(*and_args),
         )
         .where(
             and_(
                 origin_location.country == tree_location.country,
-                origin_location.division == None,  # noqa: E711
-                origin_location.location == None,  # noqa: E711
+                origin_location.division == tree_location.division,  # noqa: E711
+                origin_location.location == tree_location.location,  # noqa: E711
             )
         )
         .order_by(asc("distance"))
         .limit(15)
```

Comment:
(you prob figured it out already. just adding notes in case helpful) i think the query could benefit from a deduplication based on the `category` column (for example, `Location.division`), and in best-case scenario keep the one entry that has the smallest `distance` among duplicates. So when `category=division`, we don't end up w 15 entries w identical `division` which then become 1 color. 
This way we would always end up with 15 entries from the sql query and no need to fill extra ones below. On the tree many of the colored locations had no lontitude/latitude data (from the map under the tree) so they are likely not pulled from the database from the sql query and instead filled with the locations on the tree, indicating the query is not pulling enough locations.